tests/vmcheck/test-layering-basic.sh

```diff
@@ -108,3 +108,28 @@ vm_assert_status_jq '.deployments|length == 2'
 vm_rpmostree cleanup -pr
 vm_assert_status_jq '.deployments|length == 1'
 echo "ok cleanup"
+
+# install foo and make sure it was imported
+vm_rpmostree install foo | tee output.txt
+assert_file_has_content output.txt '^Importing:'
+
+# upgrade with same foo in repos --> shouldn't re-import
+vm_cmd ostree commit -b vmcheck --tree=ref=vmcheck
+vm_rpmostree upgrade | tee output.txt
+assert_not_file_has_content output.txt '^Importing:'
+
+# upgrade with different foo in repos --> should re-import
+vm_cmd ostree commit -b vmcheck --tree=ref=vmcheck
+# this is a bit hacky: rpm building is normally handled by make, on which
+# vmcheck itself is dependent
```

Comment:
Yeah, not proud of that one. 😶 Definitely would make sense to move RPM building into the tests directly. The automake goo would need to be made even more complex to support e.g. building multiple `foo` RPMs with different EVRs (which would be really useful in our tests).